.github/workflows/deploy.yml

```diff
@@ -0,0 +1,147 @@
+name: Deploy
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "v*"
+
+jobs:
+  # Publish starship to Crates.io
+  cargo_publish:
+    name: Publish Cargo Package
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup | Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup | Rust
+        uses: ATiltedTree/setup-rust@v1
```

Comment:
Is there anything specific lacking from `actions-rs/toolchain`?
I have no doubt that your action works perfectly, but I would prefer to stick with something widely known and used if it does the job as expected.

Reply:
I created my own version and changed to it because while `actions-rs/toolchain` works perfectly you always have to use `actions-rs/cargo` if you want annotations in your code.
Also `actions-rs/cargo` might actually be the cause of the random check failure as I have never seen it fail with my current changes.

But I can always change it back if you want to.

Reply:
We could keep using `ATiltedTree/setup-rust` to see if it solves the random failures, but I'm fairly confident that the flakiness of our tests is due to race conditions and brittleness in how our tests work.
If we continue to see flakiness, I think it'd be best if we brought it back to `actions-rs/toolchain`.